### PR TITLE
new audio APIs

### PR DESCRIPTION
**Pull request by sasxianhu**
1. Added client audio APIs to load any type of audio files
2. Enriched audio table to support audio classification
3. Fixed some bugs in model and metrics

It passes these tests:
test_audio.py
test_model.py
test_speech.py
test_image.py
test_metrics.py

### CODE REVIEW

File: dlpy/model.py [hunk 1]
```diff
@@ -271,11 +273,17 @@ def fit(self, data, inputs=None, target=None, data_specs=None, mini_batch_size=1
         # check whether the field is none or not
         if self.model_weights is not None and self.model_weights.to_table_params()['name'].upper() in \
                 list(self._retrieve_('table.tableinfo').TableInfo.Name):
-            print('NOTE: Training based on existing weights.')
-            init_weights = self.model_weights
+            if train_from_scratch:
+                print('NOTE: Ignoring the existing weights and training from scratch.')
+                init_weights = self.model_weights
```
Comment (ethem-kinginthenorth) [inline dlpy/model.py]: don't we initialize init_weights with model_weights in this way. so it still uses the trained weights table if there is any.

File: dlpy/model.py [hunk 2]
```diff
@@ -910,17 +927,29 @@ def evaluate(self, data, text_parms=None, layer_out=None, layers=None, gpu=None,
...
 
-            self.valid_res_tbl = self.conn.CASTable(valid_res_tbl)
             temp_columns = self.valid_res_tbl.columns.tolist()
-            columns = [item for item in temp_columns if item[0:9] == 'P_' + self.target or item == 'I_' + self.target]
-            img_table = self._retrieve_('image.fetchimages', fetchimagesvars=columns, imagetable=self.valid_res_tbl, to=1000)
-            img_table = img_table.Images
 
-            self.valid_res = img_table
+            # the model might not use the image data
+            doImage=False
```
Comment (ethem-kinginthenorth) [inline dlpy/model.py]: can we pythonize the variable name

File: dlpy/speech_utils.py [hunk 3]
```diff
@@ -399,3 +401,407 @@ def clean_audio(listing_path_local, segment_path_local_list):
...
+    DLPyError
+        If anything goes wrong, it complains and prints the appropriate message.
+
+    '''
+
+    try:
+        import soundfile as sf
+        import sounddevice as sd
+    except (ModuleNotFoundError, ImportError):
+        raise DLPyError('cannot import soundfile or sounddevice')
```
Comment (ethem-kinginthenorth) [inline dlpy/speech_utils.py]: can we please add one more note indicating that they might need to add those packages

File: dlpy/model.py (hunk 2 above)
Comment (sasxianhu) [reply dlpy/model.py]: Fixed it.

**Pull request merged**